src/main/java/com/example/demo/service/TeamService.java

```diff
@@ -0,0 +1,119 @@
+package com.example.demo.service;
+
+import com.example.demo.dto.Employee;
+import com.example.demo.dto.Team;
+import com.example.demo.entities.DepartmentEntity;
+import com.example.demo.entities.EmployeeEntity;
+import com.example.demo.entities.TeamEntity;
+import com.example.demo.exeption.DepartmentNotFoundException;
+import com.example.demo.exeption.EmployeeNotFoundException;
+import com.example.demo.exeption.TeamNotFoundException;
+import com.example.demo.repository.DepartmentRepository;
+import com.example.demo.repository.EmployeeRepository;
+import com.example.demo.repository.TeamRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+@Service
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public class TeamService {
+    private final TeamRepository teamRepository;
+    private final EmployeeRepository employeeRepository;
+    private final DepartmentRepository departmentRepository;
+
+    public Team saveTeam(Team team) {
+        TeamEntity teamEntity = new TeamEntity();
+        setTeamEntity(team, teamEntity);
+        TeamEntity teamEntity1 = teamRepository.save(teamEntity);
+        return convertTeamEntityToTeam(teamEntity1);
+    }
+
+    public List<Team> getAllTeams() {
+        return StreamSupport.stream(teamRepository.findAll().spliterator(), false)
+                .map(this::convertTeamEntityToTeam).collect(Collectors.toList());
+    }
+
+    public Team getTeamById(Long id) {
+        TeamEntity teamEntity = getTeamEntity(id);
+        return convertTeamEntityToTeam(teamEntity);
+    }
+
+    public Team updateTeam(Team team, Long id) {
+        TeamEntity teamEntity = setTeamEntityDetails(team, id);
+        teamRepository.save(teamEntity);
+        return convertTeamEntityToTeam(teamEntity);
+    }
+
+    public Team deleteTeam(Long id) {
+        TeamEntity teamEntity = getTeamEntity(id);
+        Team team = convertTeamEntityToTeam(teamEntity);
+        teamRepository.deleteById(id);
+        return team;
+    }
+
+    private Team convertTeamEntityToTeam(TeamEntity teamEntity) {
```

Comment:
I prefer to create a method for each case instead of using this function with so many `if` + you don't have tests for all of them